lib/algoliasearch-rails.rb

```diff
@@ -146,18 +146,18 @@ def get_default_attributes(object)
     end
 
     def get_attribute_names(object)
-      get_attributes(object).keys
+      get_unresolved_attributes(object).keys
     end
 
     def attributes_to_hash(attributes, object)
       if attributes
-        Hash[attributes.map { |name, value| [name.to_s, value.call(object) ] }]
+        Hash[attributes.map { |name, value| [name.to_s, value.is_a?(Proc) ? value.call(object) : value] }]
       else
         {}
       end
     end
 
-    def get_attributes(object)
+    def get_unresolved_attributes(object)
```

Comment:
`line 164` would still resolve your attributes. should be changed as `@serializer._attributes` to get attribute names without actually calculating the values.